In [33]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

df = pd.read_csv('./Datasets/Data_Combinada.csv')

/tmp/ipykernel_26026/3208406358.py:5: DtypeWarning: Columns (15,16) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('./Datasets/Data_Combinada.csv')


In [34]:
df.head(10)

num_expediente       fecha     hora  \
0    2022S000001  01/01/2022  1:30:00   
1    2022S000001  01/01/2022  1:30:00   
2    2022S000002  01/01/2022  0:30:00   
3    2022S000002  01/01/2022  0:30:00   
4    2022S000002  01/01/2022  0:30:00   
5    2022S000003  01/01/2022  1:50:00   
6    2022S000003  01/01/2022  1:50:00   
7    2022S000004  01/01/2022  3:40:00   
8    2022S000004  01/01/2022  3:40:00   
9    2022S000005  01/01/2022  4:30:00   

                                 localizacion   numero  cod_distrito  \
0                          AVDA. ALBUFERA, 19       19          13.0   
1                          AVDA. ALBUFERA, 19       19          13.0   
2  PLAZA. CANOVAS DEL CASTILLO / PASEO. PRADO        2           3.0   
3  PLAZA. CANOVAS DEL CASTILLO / PASEO. PRADO        2           3.0   
4  PLAZA. CANOVAS DEL CASTILLO / PASEO. PRADO        2           3.0   
5                      CALL. SAN BERNARDO, 53       53           1.0   
6                      CALL. SAN BERNARDO, 53       53           1.0   
7            CALL. ALCALA / PLAZA. EISENHOWER      728          20.0   
8            CALL. ALCALA / PLAZA. EISENHOWER      728          20.0   
9                        AUTOV. A-42, +00500E  +00500E          12.0   

              distrito                tipo_accidente estado_meteorológico  \
0   PUENTE DE VALLECAS                       Alcance            Despejado   
1   PUENTE DE VALLECAS                       Alcance            Despejado   
2               RETIRO       Colisión fronto-lateral                  NaN   
3               RETIRO       Colisión fronto-lateral                  NaN   
4               RETIRO       Colisión fronto-lateral                  NaN   
5               CENTRO           Atropello a persona            Despejado   
6               CENTRO           Atropello a persona            Despejado   
7  SAN BLAS-CANILLEJAS  Choque contra obstáculo fijo            Despejado   
8  SAN BLAS-CANILLEJAS  Choque contra obstáculo fijo            Despejado   
9                USERA  Choque contra obstáculo fijo            Despejado   

             tipo_vehiculo  ...       rango_edad sexo cod_lesividad  \
0                  Turismo  ...  De 30 a 34 años    M           NaN   
1                  Turismo  ...  De 45 a 49 años    H           NaN   
2  Motocicleta hasta 125cc  ...  De 30 a 34 años    H           NaN   
3  Motocicleta hasta 125cc  ...  De 35 a 39 años    M           NaN   
4                  Turismo  ...  De 40 a 44 años    H           NaN   
5      Motocicleta > 125cc  ...      Desconocido    H           NaN   
6      Motocicleta > 125cc  ...  De 18 a 20 años    M           7.0   
7                  Turismo  ...  De 50 a 54 años    H           2.0   
8                  Turismo  ...  De 55 a 59 años    M           3.0   
9                  Turismo  ...  De 45 a 49 años    H           NaN   

                                           lesividad coordenada_x_utm  \
0                                                NaN       443359,226   
1                                                NaN       443359,226   
2                                                NaN       441155,351   
3                                                NaN       441155,351   
4                                                NaN       441155,351   
5                                                NaN       439995,351   
6  Asistencia sanitaria sólo en el lugar del acci...       439995,351   
7                Ingreso inferior o igual a 24 horas       449693,925   
8                        Ingreso superior a 24 horas       449693,925   
9                                                NaN       438952,303   

  coordenada_y_utm positiva_alcohol  positiva_droga  numero_pasajeros  fugado  
0      4472082,272              0.0             0.0                 2      No  
1      4472082,272              0.0             0.0                 2      No  
2      4474129,588              1.0             0.0                 3      No  
3  

Procedemos a separar norte de sur

In [35]:
unique_districts = df.groupby('distrito').count().reset_index()
unique_districts['distrito']

0              ARGANZUELA
1                 BARAJAS
2             CARABANCHEL
3                  CENTRO
4               CHAMARTÍN
5                CHAMBERÍ
6           CIUDAD LINEAL
7     FUENCARRAL-EL PARDO
8               HORTALEZA
9                  LATINA
10        MONCLOA-ARAVACA
11              MORATALAZ
12     PUENTE DE VALLECAS
13                 RETIRO
14              SALAMANCA
15    SAN BLAS-CANILLEJAS
16                 TETUÁN
17                  USERA
18              VICÁLVARO
19      VILLA DE VALLECAS
20             VILLAVERDE
Name: distrito, dtype: object

In [36]:
distritos_norte = pd.DataFrame()
distritos_norte['distrito'] = ['FUENCARRAL-EL PARDO', 'MONCLOA-ARAVACA', 'TETUÁN', 'CHAMARTÍN', 'HORTALEZA',
                               'CIUDAD LINEAL', 'BARAJAS', 'CHAMBERÍ', 'SALAMANCA', 'SAN BLAS-CANILLEJAS']

In [40]:
distritos_sur = pd.DataFrame()
distritos_sur['distrito'] = ['CENTRO', 'RETIRO', 'MORATALAZ', 'VICÁLVARO', 'LATINA', 'CARABANCHEL', 'USERA',
                             'VILLA DE VALLECAS', 'VILLAVERDE', 'ARGANZUELA', 'PUENTE DE VALLECAS']

In [41]:
distritos_norte

distrito
0  FUENCARRAL-EL PARDO
1      MONCLOA-ARAVACA
2               TETUÁN
3            CHAMARTÍN
4            HORTALEZA
5        CIUDAD LINEAL
6              BARAJAS
7             CHAMBERÍ
8            SALAMANCA
9  SAN BLAS-CANILLEJAS

In [42]:
distritos_sur

distrito
0               CENTRO
1               RETIRO
2            MORATALAZ
3            VICÁLVARO
4               LATINA
5          CARABANCHEL
6                USERA
7    VILLA DE VALLECAS
8           VILLAVERDE
9           ARGANZUELA
10  PUENTE DE VALLECAS

In [79]:
df_norte = df[df['distrito'].isin(distritos_norte['distrito'])].copy()
df_sur = df[df['distrito'].isin(distritos_sur['distrito'])].copy()

Procedemos a contar el número de accidentes. Para ello no necesitamos la información del lugar ya que está demasiado granulado para lo que queremos hacer

In [80]:
df_norte.drop(
    columns=['localizacion', 'coordenada_y_utm', 'coordenada_x_utm', 'cod_distrito', 'lesividad', 'tipo_vehiculo',
             'tipo_persona', 'tipo_vehiculo', 'fugado', 'estado_meteorológico', 'numero_pasajeros', 'rango_edad',
             'numero', 'sexo', 'cod_lesividad', 'tipo_accidente', 'hora'], inplace=True)
df_sur.drop(
    columns=['localizacion', 'coordenada_y_utm', 'coordenada_x_utm', 'cod_distrito', 'lesividad', 'tipo_vehiculo',
             'tipo_persona', 'tipo_vehiculo', 'fugado', 'estado_meteorológico', 'numero_pasajeros', 'rango_edad',
             'numero', 'sexo', 'cod_lesividad', 'tipo_accidente', 'hora'], inplace=True)

In [75]:
df_norte

num_expediente       fecha      hora             distrito  \
7        2022S000004  01/01/2022   3:40:00  SAN BLAS-CANILLEJAS   
8        2022S000004  01/01/2022   3:40:00  SAN BLAS-CANILLEJAS   
11       2022S000006  01/01/2022   5:15:00             CHAMBERÍ   
12       2022S000006  01/01/2022   5:15:00             CHAMBERÍ   
14       2022S000009  01/01/2022   5:20:00  SAN BLAS-CANILLEJAS   
...              ...         ...       ...                  ...   
74954    2023S028337  31/08/2023  23:20:00            HORTALEZA   
74955    2023S028337  31/08/2023  23:20:00            HORTALEZA   
74956    2023S028337  31/08/2023  23:20:00            HORTALEZA   
74959    2023S028352  31/08/2023  14:40:00            CHAMARTÍN   
74960    2023S029136  02/08/2023  22:10:00  FUENCARRAL-EL PARDO   

       positiva_alcohol  positiva_droga  
7                   0.0             0.0  
8                   0.0             0.0  
11                  0.0             0.0  
12                  0.0             0.0  
14                  0.0             0.0  
...                 ...             ...  
74954               0.0             0.0  
74955               0.0             0.0  
74956               0.0             0.0  
74959               0.0             0.0  
74960               0.0             0.0  

[38197 rows x 6 columns]

In [84]:
expedientes_por_distrito_norte = df_norte.groupby(['distrito'])['num_expediente'].nunique().reset_index()
expedientes_por_distrito_sur = df_sur.groupby(['distrito'])['num_expediente'].nunique().reset_index()

In [93]:
std_norte = expedientes_por_distrito_norte['num_expediente'].std()
mean_norte = expedientes_por_distrito_norte['num_expediente'].mean()
median_norte = expedientes_por_distrito_norte['num_expediente'].median()
skewness_norte = expedientes_por_distrito_norte['num_expediente'].skew()
kurtosis_norte = expedientes_por_distrito_norte['num_expediente'].kurt()
q1_norte = expedientes_por_distrito_norte['num_expediente'].quantile(0.25)
q3_norte = expedientes_por_distrito_norte['num_expediente'].quantile(0.75)

std_sur = expedientes_por_distrito_sur['num_expediente'].std()
mean_sur = expedientes_por_distrito_sur['num_expediente'].mean()
median_sur = expedientes_por_distrito_sur['num_expediente'].median()
skewness_sur = expedientes_por_distrito_sur['num_expediente'].skew()
kurtosis_sur = expedientes_por_distrito_sur['num_expediente'].kurt()
q1_sur = expedientes_por_distrito_sur['num_expediente'].quantile(0.25)
q3_sur = expedientes_por_distrito_sur['num_expediente'].quantile(0.75)

combined_data = pd.concat([expedientes_por_distrito_norte['num_expediente'], expedientes_por_distrito_sur['num_expediente']])
std_combined = combined_data.std()
mean_combined = combined_data.mean()
median_combined = combined_data.median()
skewness_combined = combined_data.skew()
kurtosis_combined = combined_data.kurt()
q1_combined = combined_data.quantile(0.25)
q3_combined = combined_data.quantile(0.75)

print(f"Conjunto Norte:")
print(f"Desviación estándar: {std_norte}, Media: {mean_norte}, Mediana: {median_norte}")
print(f"Asimetría: {skewness_norte}, Curtosis: {kurtosis_norte}")
print(f"Primer Cuartil (Q1): {q1_norte}, Tercer Cuartil (Q3): {q3_norte}\n")

print(f"Conjunto Sur:")
print(f"Desviación estándar: {std_sur}, Media: {mean_sur}, Mediana: {median_sur}")
print(f"Asimetría: {skewness_sur}, Curtosis: {kurtosis_sur}")
print(f"Primer Cuartil (Q1): {q1_sur}, Tercer Cuartil (Q3): {q3_sur}\n")

print(f"Conjunto Combinado:")
print(f"Desviación estándar: {std_combined}, Media: {mean_combined}, Mediana: {median_combined}")
print(f"Asimetría: {skewness_combined}, Curtosis: {kurtosis_combined}")
print(f"Primer Cuartil (Q1): {q1_combined}, Tercer Cuartil (Q3): {q3_combined}")

Conjunto Norte:
Desviación estándar: 511.271454317567, Media: 1713.5, Mediana: 1733.0
Asimetría: -0.7267803156878198, Curtosis: 0.9957888310995626
Primer Cuartil (Q1): 1481.75, Tercer Cuartil (Q3): 2042.75

Conjunto Sur:
Desviación estándar: 577.2443156931041, Media: 1483.0, Mediana: 1504.0
Asimetría: 0.1802089413362764, Curtosis: -0.6854804901558929
Primer Cuartil (Q1): 1056.5, Tercer Cuartil (Q3): 1852.5

Conjunto Combinado:
Desviación estándar: 546.0310343526187, Media: 1592.7619047619048, Mediana: 1604.0
Asimetría: -0.22328235430724933, Curtosis: -0.6467812668454891
Primer Cuartil (Q1): 1329.0, Tercer Cuartil (Q3): 2092.0


Según la información calculada sobre las desviaciones típicas podemos realizar las siguientes conclusiones:
1. Existe una mayor tasa de accidentalidad en los distritos del norte, cuya media es más alta, sin embargo, la dispersión del conjunto es algo menos que la accidentalidad en el sur. También podemos concluir que la accidentalidad en el norte es más uniforme que la del sur, que está más dispersa respecto de la media.
2. Los datos de ambos conjuntos están bastante dispersos alrededor de la media
3. Calculamos los intervalos de confianza dentro de los cuales podemos esperar encontrar la verdadera media de la población con un 95% de confianza. Recordamos que conocemos el tamaño de muestra de ambos conjuntos.
    * Norte (10 distritos): Intervalo de confianza 95%: (1396.62, 2030.38) Amplitud del intervalo: 633.76
        *  El 50% de los datos están entre Q1: 1481.75 y Q3: 2042.75. La distancia entre Q1 y Q3 también da una idea de la dispersión de los datos.
    * Sur (11 distritos): Intervalo de confianza 95%: (1141.88, 1824.12) Amplitud del intervalo: 682.24
        * El 50% de los datos están entre estos Q1: 1056.5, Q3: 1852.5 valores. El rango intercuartílico también es bastante amplio, sugiriendo una dispersión significativa.
    * Ambos conjuntos (21 distritos en total) 95%: Intervalo de confianza: (1359.22, 1826.30)
4. La mediana es aproximadamente igual en ambos casos, lo que indica una distribución bastante simétrica
5. El conjunto norte tiene una ligera asimetría hacia la izquierda con una cola más larga y el conjunto sur tiene una ligera asimetría a la derecha
6. El conjunto norte tiene una distribución más cercana a la normalidad en términos de curtosis que el conjunto sur. El conjunto sur muestra una distribución con colas significativamente más ligeras y un pico más plano, indicando una menor presencia de valores extremos en comparación con una distribución normal y también en comparación con el conjunto norte.

En resumen el conjunto sur tiene una mayor dispersión general de valores (mayor std), pero al mismo tiempo, tiene colas más ligeras y un pico más plano (menor curtosis), lo que implica que tiene menos valores extremos. Esto podría significar que, aunque los valores en el sur varían más de la media, no tienden a alcanzar valores tan extremos como podrían ser en el norte.